### PR TITLE
Use latest WP image for local env; create database during env setup

### PR DESCRIPTION
**Pull request by pierlon**
## Summary

Previously #5974

As noted in https://github.com/docker-library/docs/pull/1892#issuecomment-797790551, the intention of the WP image was not to use default values to connect to the `mysql` container and create the database if it did not exist. Since that is now the case, in order to use the latest WP image we will have to create the database ourselves.

This PR adds the functionality to wait for the `mysql` container to fully start, and then create the database so that WP can be installed.

### CODE REVIEW

File: bin/local-env/docker-compose.yml [hunk 1]
```diff
@@ -26,6 +20,8 @@ services:
     volumes:
       - wordpress_data:/var/www/html
       - ../../:/var/www/html/wp-content/plugins/amp
+    env_file:
```
Comment (pierlon) [inline bin/local-env/docker-compose.yml:23]: Due to how the WP image is now setting the [supplied environment variables](https://github.com/docker-library/wordpress/blob/fb920c697693bad49582c3a0c8bf1fa0e7642d2b/wp-config-docker.php#L41-L56), the CLI container will no longer be able to read the correct WP constants unless the environment variables are also supplied to that container. The alternative would be to create our own `wp-config.php` file with the preset values and mount it in the WP container.

File: sizereport.config.js [hunk 2]
```diff
@@ -1,6 +0,0 @@
-module.exports = {
```
Comment (pierlon) [inline sizereport.config.js:1]: This change is unrelated to the PR. This file would configure the [size report](https://github.com/GoogleChromeLabs/travis-size-report) for Travis, which we no longer use and so this file is no longer needed.

**Review by westonruter — APPROVED** [review]
Nice work.

File: bin/local-env/install-wordpress.sh [hunk 3]
```diff
@@ -17,15 +17,26 @@ dc up -d >/dev/null 2>&1
...
+echo -en $(status_message "Waiting for database connection...")
+until $(container bash -c "echo -n > /dev/tcp/mysql/3306" >/dev/null 2>&1); do
+    echo -n '.'
+    sleep 5
+done
+echo ''
+
+# Create the database if it doesn't exist.
+echo -e $(status_message "Creating the database (if it does not exist)...")
+mysql -e 'CREATE DATABASE IF NOT EXISTS wordpress;'
```
Comment (swissspidy) [inline bin/local-env/install-wordpress.sh:38]: Shouldn't this use `WORDPRESS_DB_NAME` from the env file?

**Comment by swissspidy** [comment]
@pierlon If I understand correctly, this uses a `.env` file to reduce duplication, but theoretically you could just pass `environment:` to both containers, right? I'd want to avoid adding such a file if possible :-)

File: bin/local-env/install-wordpress.sh (hunk 3 above)
Comment (pierlon) [reply bin/local-env/install-wordpress.sh:38]: Yes it should have, I'll make the amendment.

**Comment by pierlon** [comment]
@swissspidy Yes passing `environment:` to both containers works just fine :smile:.
